src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubConfiguration.cs

```diff
@@ -201,17 +216,24 @@ public void AddReceiver(string eventHubName, string receiverConnectionString, st
             };
         }
         
-        internal EventHubClient GetEventHubClient(string eventHubName)
+        internal EventHubClient GetEventHubClient(string eventHubName, string connection)
         {
             EventHubClient client;
             if (_senders.TryGetValue(eventHubName, out client))             
             {
                 return client;
             }
+            else if (!string.IsNullOrWhiteSpace(connection) && NameResolver != null)
+            {
+                var resolvedConnection = NameResolver.Resolve(connection);
```

Comment:
Misconception on my part - thought that AppSetting attribute would throw if there was no value specified. It does, assuming that 1. there is a connection set, explicitly or via default and 2. there is no app setting/env variable with that connection name.